### PR TITLE
Fixes for Dart/native conversion of some painting objects

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/engine/core/painting/Paint.dart [hunk 1]
```diff
@@ -83,15 +83,15 @@ class Paint {
     }
 
     return <dynamic>[
-      style.index,
+      style != null ? style.index : null,
```
Comment (abarth) [inline sky/engine/core/painting/Paint.dart]: `style?.index`

**Comment by abarth** [comment]
LGTM w/ the fancy `?.` operator.

File: sky/engine/core/painting/Paint.dart (hunk 1 above)
Comment (jason-simmons) [inline sky/engine/core/painting/Paint.dart]: done

**Comment by abarth** [comment]
LGTM

**Pull request merged**